libsolidity/formal/SolverInterface.h

```diff
@@ -65,16 +65,17 @@ struct FunctionSort: public Sort
 {
 	FunctionSort(std::vector<SortPointer> _domain, SortPointer _codomain):
 		Sort(Kind::Function), domain(std::move(_domain)), codomain(std::move(_codomain)) {}
-	bool operator==(FunctionSort const& _other) const
+	bool operator==(Sort const& _other) const override
 	{
+		auto _otherFunction = dynamic_cast<FunctionSort const&>(_other);
```

Comment:
We usually put `solAssert` before such casts to have the usual detailed exception.

Reply:
Even with references? This way the cast would throw an exception anyway